docs/api.rst

```diff
@@ -0,0 +1,264 @@
+.. _ksql-http-api:
+
+HTTP API Reference
+==================
+
+Content Types
+-------------
+
+The ksql HTTP server uses content types for requests and responses to indicate the serialization format of the data and the api version. Currently, the only serialization format supported is JSON. The only version supported is v1. Your request should specify this serialization format and version in the ``Accept`` header as::
+
+    Accept: application/vnd.ksql.v1+json
+
+The less specific ``application/json`` content type is also permitted. However, this is only for compatibility and ease of use, and you should use the versioned value if possible.
+
+The server also supports content negotiation, so you may include multiple, weighted preferences::
+
+    Accept: application/vnd.ksql.v1+json; q=0.9, application/json; q=0.5
+
+which can be useful when, for example, a new version of the API is preferred but you cannot be certain it is avialable yet.
+
+Errors
+------
+
+All API endpoints use a standard error message format for any requests that return an HTTP status indicating an error (any 4xx or 5xx statuses):
+
+   .. sourcecode:: http
+
+      HTTP/1.1 <Error Status>
+      Content-Type: application/json
+
+      {
+          "error_code": <Error code>
+          "message": <Error Message>
+      }
+
+Some endpoints may include additional fields as well that provide more context for handling the error.
+
+Run a KSQL Statement
+--------------------
+
+The ksql resource runs a sequence of ksql statements. Any statement except those starting with ``SELECT`` can be run on this endpoint. To run ``SELECT`` statements use the ``/query`` endpoint.
+
+.. http:post:: /ksql
+
+   Run a sequence of ksql statements.
+
+   :json string ksql: A semicolon-delimited sequence of KSQL statements to run.
+   :json map streamsProperties: Property overrides to run the statements with. Refer to the :ref:`Config Reference <ksql-param-reference>` for details on properties that can be set.
+   :json string streamsProperties[<property-name>]: The value of the property named by property-name. Both the value and property-name should be strings.
+
+   The response json is an array of result objects. The contents of each result object depends on the statement for which it is returning results. The following sections detail the contents of the result objects by statement.
+
+   **CREATE, DROP, TERMINATE**
+
+   :>json string statementText: The ksql statement for which the result is being returned.
+   :>json string commandId: A string that identifies the requested operation. This id can be used to poll the result of the operation using the status endpoint.
```

Comment:
`You can use this ID to poll the result of the operation using the status endpoint.`